src/datastore/postgres-store.ts

```diff
@@ -2944,9 +2945,16 @@ export class PgDataStore
         txs = txQuery.rows.map(r => this.parseTxQueryResult(r));
       }
       if (metadata?.microblocks) {
-        const microblocksQuery = await client.query<MicroblockQueryResult>(
+        const microblocksQuery = await client.query<
+          MicroblockQueryResult & { transaction_count: number }
+        >(
           `
-          SELECT ${MICROBLOCK_COLUMNS}
+          SELECT ${MICROBLOCK_COLUMNS}, (
+            SELECT COUNT(tx_id)::integer as transaction_count
+            FROM txs
+            WHERE txs.microblock_hash = microblocks.microblock_hash
```

Comment:
This filter seems wrong... looks like you'll count every transaction that exists on the table that was included in any microblock (not just the ones from this block)

Reply:
The `WHERE` clause filters out only those microblocks that were found in the desired block. This query does return `count` for `streamed_microblocks` along with `accepted_microblocks` but txs count for streamed are ignored later. 
I have also updated a test that validates this functionality, [here](https://github.com/hirosystems/stacks-blockchain-api/pull/1162/files#diff-521b3ebceedf4b02e9f0a164d139b5b2cc65fbeae7c5a54d4f5ce265be2bb1e1R7453-R7455).
Plus, there is a query snippet you can look at [here](https://www.db-fiddle.com/f/x2cjAX38WZLNkSdavMzbkz/0).